DevTools/HackStudio/LanguageServers/Cpp/CppLanguageServer.ipc

```diff
@@ -0,0 +1,11 @@
+endpoint CppLanguageServer = 8001
+{
+    Greet(String project_root) => (i32 client_id)
+
+    FileOpened(String file_name) =|
+    FileEditInsertText(String file_name, String text, i32 start_line, i32 start_column) =|
+    FileEditRemoveText(String file_name, i32 start_line, i32 start_column, i32 end_line, i32 end_column) =|
+    SetFileContent(String file_name, String content) =|
+
+    AutoCompleteSuggestions(String file_name, i32 cursor_line, i32 cursor_column) => (Vector<String> suggestions)
```

Comment:
I hope we're not calling this synchronously on the UI thread? 😨

Reply:
Oh, we definitely are :stuck_out_tongue_closed_eyes: 
Currently auto-completion is very simple and does only lexing, so there isn't any noticeable delay in the UI thread (I mean, we perform lexing on the program every time we edit).
Of course once it starts doing more complex stuff, we would not want to call this synchronously from the UI thread.
We could use server->client IPC messages for this, and then `AutoCompleteSuggestions` could be asynchronous and report the results back to the client when we have them.
I think having this call synchronous for now is OK to get the language server going, but I added a `FIXME` in `ClientConnection.cpp` about this (I don't think our .ipc compiler supports comment?),

Reply:
Yeah, I guess we can get away with that for now (*somehow*, everything is quite fast for us to do on the UI thread — perhaps because we don't do animations and don't use the system under load a lot). But on other systems, it'd be a non-starter to do *any* blocking I/O on the UI thread, let alone RPCing into a remote process.